devSandbox.m

```diff
@@ -0,0 +1,238 @@
+% rounding the size to the nearest pixel, leaving PTB to do some scaling.
+noise = rand(round(visibleSize / 2)) .* white;
+
+% Make our noise texture
+noiseTex = Screen('MakeTexture', cfg.win, noise);
+
+% Make a destination rectangle for our textures and center this on the
+% screen
+dstRect = [0 0 visibleSize visibleSize];
+dstRect = CenterRect(dstRect, cfg.winRect);
+
+% Calculate the wait duration
+waitDuration = waitframes * cfg.ifi;
+
+% Recompute pixPerCycle, this time without the ceil() operation from above.
+% Otherwise we will get wrong drift speed due to rounding errors
+pixPerCycle = 1 / freqCyclesPerPix;
+
+% Translate requested speed of the grating (in cycles per second) into
+% a shift value in "pixels per frame"
+shiftPerFrame = cyclesPerSecond * pixPerCycle * waitDuration;
+
+% Sync us to the vertical retrace
+vbl = Screen('Flip', cfg.win);
+
+% Set the frame counter to zero, we need this to 'drift' our grating
+frameCounter = 0;
+
+% Loop until a key is pressed
+while ~KbCheck
+
+    % Calculate the xoffset for our window through which to sample our
+    % grating
+    xoffset = mod(frameCounter * shiftPerFrame, pixPerCycle);
+
+    % Now increment the frame counter for the next loop
+    frameCounter = frameCounter + 1;
+
+    % Define our source rectangle for grating sampling
+    srcRect = [xoffset 0 xoffset + visibleSize visibleSize];
+
+    % Draw noise texture to the screen
+    Screen('DrawTexture', cfg.win, noiseTex, [], dstRect, []);
+
+    % Draw grating mask
+    Screen('DrawTexture', cfg.win, gratingMaskTex, srcRect, dstRect, []);
+
+    % Flip to the screen on the next vertical retrace
+    vbl = Screen('Flip', cfg.win, vbl + (waitframes - 0.5) * cfg.ifi);
+
+end
+
+
+
+% -------------------------------------------------------------------------
+%%
+
+% Close PTB, see the Sub-Functions below
+devSandbox_cleanUp
+
+catch
+
+    devSandbox_cleanUp
+    psychrethrow(psychlasterror);
+
+end
+
+
+end
+
+%% Sub-Functions
+function cfg = devSandbox_initPTB(cfg)
+
+% Shorter version of `initPTB.m`
+
+% Skip the PTB sync test
+Screen('Preference', 'SkipSyncTests', 2);
+
+% Open a transparent window
+PsychDebugWindowConfiguration
+
+% Here we call some default settings for setting up Psychtoolbox
+PsychDefaultSetup(2);
+
+% Get the screen numbers and draw to the external screen if avaliable
+cfg.screen = max(Screen('Screens'));
+
+% Open an on screen window
+[cfg.win, cfg.winRect] = Screen('OpenWindow', cfg.screen, cfg.backgroundColor);
+
+% Get the size of the on screen window
+[cfg.winWidth, cfg.winHeight] = WindowSize(cfg.win);
+
+% Query the frame duration
+cfg.ifi = Screen('GetFlipInterval', cfg.win);
+
+% Get the Center of the Screen
+cfg.center = [cfg.winRect(3), cfg.winRect(4)]/2;
+
+% Set up alpha-blending for smooth (anti-aliased) lines
+Screen('BlendFunction', cfg.win, 'GL_SRC_ALPHA', 'GL_ONE_MINUS_SRC_ALPHA');
+
+end
+
+function devSandbox_cleanUp
```

Comment:
You prefer to have a subfunction here rather than a using the default "cleanUp" so that everything is self contained?

Reply:
yes so that, being a stand-alone, I can copy-paste it in my experiment folder and do version control there